ipalib/errors.py

```diff
@@ -1697,6 +1697,15 @@ class CertificateFormatError(CertificateError):
     format = _('Certificate format error: %(error)s')
 
 
+class CertificateMappingError(CertificateError):
```

Comment:
1. You can subclass `SingleMatchExpected` with your own format string.
2. I would use a new `ExecutionError` subclass for this.
3. `RequirementError` is not a good choice, as it is a validation error, but what you are describing is an execution error.